### PR TITLE
Add pattern: Services - Three Columns

### LINKED ISSUE #80: Block Patterns - Services - Three columns with headings and paragraphs
**Description**

🛠️ The goal of this issue is to track the implementation of the `Services - Three-columns with headings and paragraphs` pattern. 
🧑‍💻 The implementation must follow the [pattern creation guidelines](https://github.com/WordPress/twentytwentyfive/blob/trunk/README.md#pattern-creation-guidelines), and should be assigned to its appropriate category.
🎨 [Figma design](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3177-1805&t=IqDHO3nwfV13Hcvm-1)

Services_ Three columns with headings and paragraphs

### PR DESCRIPTION
**Pull request by huzaifaalmesbah**
Fixes #80

### CODE REVIEW

File: patterns/services-three-columns.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Description: Displays three columns with images and text for services.
+ *
+ * @package WordPress
+ * @subpackage Twenty_Twenty_Five
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"right":"var:preset|spacing|50","left":"var:preset|spacing|50","top":"var:preset|spacing|50","bottom":"var:preset|spacing|50"},"blockGap":"var:preset|spacing|50"}},"backgroundColor":"base","layout":{"type":"constrained"}} -->
```
Comment (juanfra) [inline patterns/services-three-columns.php]: I believe we can avoid using the background color when it's base.

File: patterns/services-three-columns.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+<div class="wp-block-group alignfull has-base-background-color has-background" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:heading -->
+<h2 class="wp-block-heading">Our services</h2>
+<!-- /wp:heading -->
+
+<!-- wp:columns -->
+<div class="wp-block-columns"><!-- wp:column -->
+<div class="wp-block-column"><!-- wp:image {"aspectRatio":"4/3","scale":"cover","sizeSlug":"full"} -->
+<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images//image-from-rawpixel-id-2222755.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```
Comment (juanfra) [inline patterns/services-three-columns.php]: There's a double slash in the image URL.

```suggestion
<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2222755.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```

File: patterns/services-three-columns.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+<!-- /wp:heading -->
+
+<!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size">Like flowers that bloom in unexpected places, every story unfolds with beauty and resilience</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+
+<!-- wp:column -->
+<div class="wp-block-column"><!-- wp:image {"aspectRatio":"4/3","scale":"cover","sizeSlug":"full","className":"is-style-default"} -->
+<figure class="wp-block-image size-full is-style-default"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images//image-from-rawpixel-id-2224378.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```
Comment (juanfra) [inline patterns/services-three-columns.php]: There's a double slash in the image URL.
```suggestion
<figure class="wp-block-image size-full is-style-default"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2224378.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```

File: patterns/services-three-columns.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+<!-- /wp:heading -->
+
+<!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size">Like flowers that bloom in unexpected places, every story unfolds with beauty and resilience</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+
+<!-- wp:column -->
+<div class="wp-block-column"><!-- wp:image {"aspectRatio":"4/3","scale":"cover","sizeSlug":"full"} -->
+<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images//image-from-rawpixel-id-2211732.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```
Comment (juanfra) [inline patterns/services-three-columns.php]: There's a double slash in the image URL.
```suggestion
<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2211732.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```

**Review by juanfra — COMMENTED** [review]
Thanks for the PR @huzaifaalmesbah!

I left a few comments on the code, to improve some things. I'd like to ask is if we can work on the indentation of the code. And also, if we can include artifacts when we're adding PRs that include changes that are visual (screenshot or screencast will work). Thanks 🙌

**Comment by huzaifaalmesbah** [comment]
Thank you for reviewing, @juanfra. I made the changes according to your suggestion.

File: patterns/services-three-columns.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"right":"var:preset|spacing|50","left":"var:preset|spacing|50","top":"var:preset|spacing|50","bottom":"var:preset|spacing|50"},"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group alignfull" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:heading -->
+<h2 class="wp-block-heading">Our services</h2>
+<!-- /wp:heading -->
+
+<!-- wp:columns -->
+<div class="wp-block-columns"><!-- wp:column -->
+<div class="wp-block-column"><!-- wp:image {"aspectRatio":"4/3","scale":"cover","sizeSlug":"full"} -->
+<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2222755.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```
Comment (shail-mehta) [inline patterns/services-three-columns.php]: @huzaifaalmesbah alt text are missing for the images. Could you please add them to all three images?

**Comment by shail-mehta** [comment]
Alt tags are missing for the images. Could you please add them to all three images in Line no 23,37 & 51

**Comment by rejaulalomkhan** [comment]
Thanks for the PR @huzaifaalmesbah 
It worked excellently in my tests.

File: patterns/services-three-columns.php (hunk 5 above)
Comment (juanfra) [inline patterns/services-three-columns.php]:
```suggestion
<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2222755.webp" alt="Image for service" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```

File: patterns/services-three-columns.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+<!-- /wp:heading -->
+
+<!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size">Like flowers that bloom in unexpected places, every story unfolds with beauty and resilience</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+
+<!-- wp:column -->
+<div class="wp-block-column"><!-- wp:image {"aspectRatio":"4/3","scale":"cover","sizeSlug":"full","className":"is-style-default"} -->
+<figure class="wp-block-image size-full is-style-default"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2224378.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```
Comment (juanfra) [inline patterns/services-three-columns.php]:
```suggestion
<figure class="wp-block-image size-full is-style-default"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2224378.webp" alt="Image for service" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```

File: patterns/services-three-columns.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+<!-- /wp:heading -->
+
+<!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size">Like flowers that bloom in unexpected places, every story unfolds with beauty and resilience</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+
+<!-- wp:column -->
+<div class="wp-block-column"><!-- wp:image {"aspectRatio":"4/3","scale":"cover","sizeSlug":"full"} -->
+<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2211732.webp" alt="" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```
Comment (juanfra) [inline patterns/services-three-columns.php]:
```suggestion
<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/image-from-rawpixel-id-2211732.webp" alt="Image for service" style="aspect-ratio:4/3;object-fit:cover"/></figure>
```

File: patterns/services-three-columns.php [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+ * Categories: call-to-action, banner, featured, services
+ * Description: Displays three columns with images and text for services.
+ *
+ * @package WordPress
+ * @subpackage Twenty_Twenty_Five
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+
```
Comment (juanfra) [inline patterns/services-three-columns.php:19]: Let's remove this empty line.

**Review by juanfra — CHANGES_REQUESTED** [review]
Thanks for working on this @huzaifaalmesbah 

It looks like we need to do a few adjustments to the layout. And also add the `alt` to the images. Please check the following screencast

https://github.com/user-attachments/assets/3848a576-f180-440c-9a30-b2e0306e2d5b

**Comment by huzaifaalmesbah** [comment]
Hello @juanfra, 
Thank you for your review and suggestions. I’ve implemented the changes based on your feedback. Could you please review my updated PR?

Thanks!

**Review by juanfra — CHANGES_REQUESTED** [review]
Thank you for working on this @huzaifaalmesbah 

I still see that the layout needs to respect the width of the designs.

https://github.com/user-attachments/assets/97a1c604-bbfe-486b-8f57-f6866f2ce80c

On [this comment](https://github.com/WordPress/twentytwentyfive/pull/114#pullrequestreview-2263294600), I showed a way you could potentially fix it. I also see that the code needs indentation.

Can you please work on these things? Thanks in advance.

**Comment by huzaifaalmesbah** [comment]
https://github.com/user-attachments/assets/0537eaac-e336-448f-8ab6-befcc1821b14

**Review by juanfra — CHANGES_REQUESTED** [review]
Thanks for the changes @huzaifaalmesbah 

The layout looks good now. Can you please work on the indentation? Once that's fixed this one should be good to go.

**Review by juanfra — APPROVED** [review]
Thank you very much @huzaifaalmesbah 🏅

**Pull request merged**